src/guides/building-a-tic-tac-toe-game-with-truffle-and-unity/index.md

```diff
@@ -181,115 +139,393 @@ To end a game, we first make sure the game we're attempting to end hasn't alread
+
+### Interacting with Smart Contracts in Unity
+
+Now that we've hooked up connecting our wallet, let's walk through an example of interacting with a smart contract.
+
+When we start a game, we want to call the function `startGame` from `TicTacToe.sol`. In order to do this, we will use the generated `StartGameRequestAndWaitForReceiptAsync` function from `unity/Assets/Contracts/csharp/Contracts/TicTacToeService.cs`. We will be using this in the `InitializeGame` function in `NewGameMenu.cs`.
+
+To reference the deployed contract, we need to create an instance of of web3, as well as reference the global address we changed earlier:
+
+```csharp
+var metaMask = MetaMaskUnity.Instance;
+var web3 = metaMask.CreateWeb3();
+var ticTacToeAddress = ContractManager.ticTacToeAddress;
+
+var ticTacToe = new Truffle.Contracts.TicTacToeService(web3, ticTacToeAddress);
+```
+
+`StartGameFunction` is an autogenerated C# class found under `csharp/Functions`. We will use this to pass in the necessary parameters to the `startGame` function in `TicTacToe.sol`:
+
+```csharp
+// This is in StartGameFunction.cs
+
+public class StartGameFunctionBase : FunctionMessage
+{
+    [Parameter("address", "payout_x", 1)]
+    public virtual string PayoutX { get; set; }
+    [Parameter("address", "payout_o", 2)]
+    public virtual string PayoutO { get; set; }
+}
+```
+
+Then, as you can see back in `NewGameMenu.cs`, we pass in `startGameFunction` to the autogenerated `StartGameRequestAndWaitForReceiptAsync` function to actually call the smart contract function. Note that we want to pass in a certain amount of ETH to be added to the jackpot. We do this by adding `AmountToSend` to `startGameFunction`.
+
+```csharp
+// We create the StartGameFunction object so we can attach ETH via AmountToSend.
+
+var startGameFunction = new Truffle.Functions.StartGameFunction();
+    startGameFunction.AmountToSend = jackpotInt;
+    startGameFunction.PayoutX = playerXValue;
+    startGameFunction.PayoutO = playerOValue;
+
+try
+{
+    var receipt = await ticTacToe.StartGameRequestAndWaitForReceiptAsync(startGameFunction);
+    ...
+}
+```
+
+As we saw in `ConnectionCanvas` properties, we can also see this script being utilized in `NewGameCanvas`:
+
+<img src="./img/newgamecanvas.png" alt="NewGameCanvas" width="40%" height="20%"/>
+
+And the game being started via `ValidateForm` in `PlayButton` properties:
+
+<img src="./img/playbutton.png" alt="PlayButton" width="40%" height="20%"/>
+
+There are many instances where we are calling the auto-generated csharp smart contract functions. Feel free to peruse around the code to get a better feel of the usage.
+
+### Play the game
+
+That was a lot to cover! Here's a short video clip of running through this entire process. Do note that the wallet connection portion is not screen recorded since we are connecting through mobile. However, the UX flow on your phone should look like this:
```

Comment:
I think this might have been raised elsewhere but the video doesn't look like its embedded.